src/runtime_src/driver/xclng/drm/xocl/userpf/xocl_ioctl.c

```diff
@@ -98,8 +104,8 @@ int xocl_execbuf_ioctl(struct drm_device *dev,
 	int numdeps;
 	int ret = 0;
 
-	if (atomic_read(&xdev->needs_reset)) {
-		userpf_err(xdev, "device needs reset, use 'xbsak reset -h'");
+	if (xdev->needs_reset) {
```

Comment:
Don't need a lock?

Reply:
You mentioned that if the data is aligned it by default atomic.  This is a Boolean only.

Reply:
This is the 3rd case mentioned in the issue: behavior changed based on value of a variable shared between multiple threads. This usually indicates that a lock is needed. It was not clear to me why we don't use a lock here. Let's have some more discussion when I'm in office this afternoon.